### PR TITLE
Added pack and sdk docs

### PR DESCRIPTION
**Pull request by ronykoz**
## Status
Ready

Added docs update Content Packs + updated docs to mention the demisto-sdk

### CODE REVIEW

**Comment by ronykoz** [comment]
Not sure about the sidebar i should put the packs-format in, should that be in Basics?

File: docs/packs_format.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+pack, and some metadata regarding the pack itself.
+Please note that all of the following files will be created using the `demisto-sdk init --pack`, and some of them 
+will have to be filled by you. An explanation for each of them will be provided below.
+
+#### .secrets-ignore
+This file will be used while running the `demisto-sdk secrets`([explanation](https://github.com/demisto/demisto-sdk/blob/master/docs/secrets.md)), we will determine the file and will
+ use it as a  white list of approved words for your PR.
+ 
+#### metadata.json
+This file will contain all the relevant metadata about the pack and will be maintained in the future using the demisto-sdk.
```
Comment (anara123) [inline docs/packs_format.md]: they should fill it now.
They will fill only the author contact details, and support:true or false
Dump here small example

File: docs/packs_format.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+Information that you can find there is: the author name(you), pack id, etc...  
+This file will generate some of its information based on your pack files, and some of the data will have to be filled
+ by you. This data will be fields like the support details, in order to contact you about the pack you've just wrote,
+  or the pack name.
+
+#### changelog.json
```
Comment (anara123) [inline docs/packs_format.md]: there is no changelog.json just CHANGELOG.md

File: docs/testing.md [hunk 3]
```diff
@@ -16,6 +16,10 @@ We use a standard naming convention for our playbook tests which follows the for
 
 ```Integration_Name-Test```
 
+There are two option to create a Test playbook:
+1. use the `emisto-sdk generate-test-playbook` command which can be found [here](https://github.com/demisto/demisto-sdk#generate-test-playbook)
```
Comment (anara123) [inline docs/testing.md]:
```suggestion
1. use the `demisto-sdk generate-test-playbook` command which can be found [here](https://github.com/demisto/demisto-sdk#generate-test-playbook)
```

File: docs/tutorial-setup-dev.md [hunk 4]
```diff
@@ -324,7 +324,10 @@ Then, make sure that `demisto-sdk` has been installed automatically by the boots
 Use demisto-sdk -h to see the available commands.
 ```
 
-Now, run the `demisto-sdk lint` command on the folder `Integrations/HelloWorld` using the `-d` option. It will run both the [linters](linting) and [pytest](unit-testing):
+Now, run the `demisto-sdk lint` command on the folder `Integrations/HelloWorld` using the `-d` option, or if you have
+ more then one change you can use the `demisto-sdk lint -g` option which will detect the changes you've made and will
```
Comment (anara123) [inline docs/tutorial-setup-dev.md]: more then one change sounds incorrect

File: docs/tutorial-setup-dev.md [hunk 5]
```diff
@@ -394,6 +397,12 @@ Make sure you use **PascalCase** in the directory name (i.e. `MyIntegration`):
 ### Step 7: Commit and push
 
 The last step is to `commit` your changes and `push` them to the *origin* in order to make sure that the pre-commit checks work fine.
+But you can also run the hooks locally using the demisto-sdk, in order to do that you can run the commands:
+1. `demisto-sdk format` - this will auto correct couple of things in order for our validation to pass.
+2. `demisto-sdk valiadte -g` - this will validate the integrity of the yml files, and will make sure they follow
```
Comment (anara123) [inline docs/tutorial-setup-dev.md]:
```suggestion
2. `demisto-sdk validate -g` - this will validate the integrity of the yml files, and will make sure they follow 
```

File: docs/tutorial-setup-dev.md [hunk 6]
```diff
@@ -394,6 +397,12 @@ Make sure you use **PascalCase** in the directory name (i.e. `MyIntegration`):
 ### Step 7: Commit and push
 
 The last step is to `commit` your changes and `push` them to the *origin* in order to make sure that the pre-commit checks work fine.
+But you can also run the hooks locally using the demisto-sdk, in order to do that you can run the commands:
+1. `demisto-sdk format` - this will auto correct couple of things in order for our validation to pass.
+2. `demisto-sdk valiadte -g` - this will validate the integrity of the yml files, and will make sure they follow 
+our pre-set of roles. You can see the [docs](https://github.com/demisto/demisto-sdk/blob/master/docs/validate_command.md)
+3. `demisto-sdk lint -g` - this will run lint and pytest on your changed python files. You can see the [docs]
```
Comment (anara123) [inline docs/tutorial-setup-dev.md]: I think they better use -d

File: docs/getting-started-guide.md [hunk 7]
```diff
@@ -54,12 +54,16 @@ The platform comes with a rich set of features and functionality that allow for
 Please read the following guidelines. Following these guidelines will maximize the chances for a fast, easy and effective review process for everyone involved. If something is not clear, please don't hesitate to reach out to us via GitHub, [Slack](http://go.demisto.com/join-our-slack-community), or [email](mailto:info@demisto.com)
 
 * Setup a development environment by following the [Dev Setup Guide](dev-setup).
+* Use the [Pack format](packs_format)) to add your contribution. 
 * Use the [Package Directory Structure](package-dir) for all Python code. If working on existing code, beyond trivial changes, we require converting to this structure as it allows running linting, unit tests and provides a clearer review process.
```
[the remark this answers is not in the document]
Comment (fvigo) [reply docs/getting-started-guide.md]: Do we still want to leave the documentation for the older Package Directory Structure for integration for contributors that already started developing?

File: docs/tutorial-setup-dev.md [hunk 8]
```diff
@@ -324,7 +324,10 @@ Then, make sure that `demisto-sdk` has been installed automatically by the boots
 Use demisto-sdk -h to see the available commands.
 ```
 
-Now, run the `demisto-sdk lint` command on the folder `Integrations/HelloWorld` using the `-d` option. It will run both the [linters](linting) and [pytest](unit-testing):
+Now, run the `demisto-sdk lint` command on the folder `Integrations/HelloWorld` using the `-d` option, or if you have
```
Comment (fvigo) [inline docs/tutorial-setup-dev.md]: HelloWorld is moved to Packs now

File: docs/tutorial-setup-dev.md [hunk 9]
```diff
@@ -394,6 +397,12 @@ Make sure you use **PascalCase** in the directory name (i.e. `MyIntegration`):
 ### Step 7: Commit and push
 
 The last step is to `commit` your changes and `push` them to the *origin* in order to make sure that the pre-commit checks work fine.
+But you can also run the hooks locally using the demisto-sdk, in order to do that you can run the commands:
+1. `demisto-sdk format` - this will auto correct couple of things in order for our validation to pass.
```
Comment (fvigo) [inline docs/tutorial-setup-dev.md]: Can you document which ones? Or point to the demisto-sdk format doc?

File: docs/unit-testing.md [hunk 10]
```diff
@@ -113,10 +112,3 @@ Sample output:
 docker run --rm -it devtestdemisto/python:1.3-alpine-1b9f5bee16a24c3f5463e324c1bb075e sh
 ```
 
-* If you have faced the error `ValueError: unknown locale: UTF-8` when running `pkg_dev_test_tasks.py`, add these lines to your ~/.bash_profile: 
-```
-export LC_ALL=en_US.UTF-8
-export LANG=en_US.UTF-8
```
Comment (glicht) [inline docs/unit-testing.md]: Why did you delete this? Let's leave this trouble shooting tip. Just change `pkg_dev_test_tasks.py` to `demisto-sdk`.

File: docs/packs_format.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+id: packs_format
+title: Packs Structure
+---
+
+As part of the content contribution process we are using a new format, Packs.
+This new content artifact will behave like a mini content repo. we will have all the relevant content items relevant 
+fot that pack located within it.
+<br/><br/>
+For instance a pack for CortexXDR will look as follows: 
+![](doc_imgs/67718809-342b8e80-f9d9-11e9-8a08-1b8ac87ffd77.png)
```
Comment (glicht) [inline docs/packs_format.md]: I would remove the image. I don't want to update the image every time we change something small. Let's just provide a link to the directory in content repo.

File: docs/packs_format.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
+---
+id: packs_format
```
Comment (glicht) [inline docs/packs_format.md]: Change the id to packs-format and the file name. Let's keep things consistent. Also need to add the id to sidebars.

**Comment by glicht** [comment]
In general my feeling is that the two docs: **packs structure** and **Package Directory Structure** need to be merged into one doc and put in the basics section.

File: docs/howtos/integrations/packs-format.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
+---
+id: packs-format
+title: Content Packs Structure
+---
+
+As part of the content contribution process we are using a structure called `Content Packs`.  
+This content artifact will behave like a mini content repo. we will have all the relevant content items relevant
```
Comment (anara123) [inline docs/howtos/integrations/packs-format.md]:
```suggestion
This content artifact will behave like a mini content repo. We will have all the relevant content items relevant 
```

File: docs/howtos/integrations/packs-format.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+- Layouts
+- Classifiers
+- Misc
+- IndicatorFields
+- Connections
+- TestPlaybooks
+```
+
+### Pack files
+The pack will have few files for its' configuration. And will give you a place to add your documentation for the
```
Comment (anara123) [inline docs/howtos/integrations/packs-format.md]:
```suggestion
The pack will have few files for it's configuration. And will give you a place to add your documentation for the 
```

File: docs/tutorials/getting-started/tut-setup-dev.md [hunk 15]
```diff
@@ -324,8 +324,9 @@ Then, make sure that `demisto-sdk` has been installed automatically by the boots
 Use demisto-sdk -h to see the available commands.
 ```
 
-Now, run the `demisto-sdk lint` command on the folder `Integrations/HelloWorld` using the `-d` option. It will run both the [linters](../../howtos/integrations/linting) and [pytest](../../howtos/integrations/unit-testing):
-
+Now, run the `demisto-sdk lint` command on the folder `Packs/HelloWorld/Integrations/HelloWorld` using the `-d` option, or if you have
+multiple changes you can use the `demisto-sdk lint -g` option which will detect the changes you've made and will run our linter based on the changed files in your cloned repo.
```
Comment (anara123) [inline docs/tutorials/getting-started/tut-setup-dev.md]: I would change it to be something like `demisto-sdk lint -g` will run on all the committed files on your branch

Comment (anara123) [reply docs/tutorials/getting-started/tut-setup-dev.md]: multiple changes can be misleading here

File: docs/tutorials/getting-started/tut-setup-dev.md [hunk 16]
```diff
@@ -382,19 +383,80 @@ In order to create a branch, use the  `git checkout -b [branch_name]` command, w
...
 
-Make sure you use **PascalCase** in the directory name (i.e. `MyIntegration`):
+Make sure you use **PascalCase** in the directory name (i.e. `MyIntegration`), you can create a Pack and an Integration directory using the `demisto-sdk init` command.
 
+An example of creating a pack called test, with an integration called test, and with the metadata file created automatically: 
 ```bash
-(venv) sb@dddd:~/demisto/content$ mkdir Integrations/MyIntegration
-(venv) sb@dddd:~/demisto/content$ git add Integrations/MyIntegration
+➜  content-docs2 git:(add-pack-and-sdk-docs) ✗ demisto-sdk init --pack 
+Please input the name of the initialized pack: test
```
Comment (anara123) [inline docs/tutorials/getting-started/tut-setup-dev.md]: not good example for partner tutorial.
it should be as if partner run this command.
`Support type of the pack:` is not option `1` but `2`
insert meaningful names , you can use existing partner integration or not.
But insert contact details, url to support, etc.
`Server min version: 5.0.0`

File: docs/tutorials/getting-started/tut-setup-dev.md [hunk 17]
```diff
@@ -382,19 +383,80 @@ In order to create a branch, use the  `git checkout -b [branch_name]` command, w
...
+Finished creating integration: test/Integrations/test.
+
 ```
 
 ### Step 7: Commit and push
 
 The last step is to `commit` your changes and `push` them to the *origin* in order to make sure that the pre-commit checks work fine.
 
+But you can also run the hooks locally using the demisto-sdk, in order to do that you can run the commands:  
+1. `demisto-sdk format` - this will auto correct couple of things in order for our validation to pass.
```
Comment (anara123) [inline docs/tutorials/getting-started/tut-setup-dev.md]: add link to full format docs

**Comment by ronykoz** [comment]
@anara123 i merged from master, and it seems to run smoothly on my local env
image

**Comment by fvigo** [comment]
Great, is it ready for merge @glicht ?
